datafusion/src/physical_plan/windows.rs

```diff
@@ -412,11 +412,14 @@ impl ExecutionPlan for WindowAggExec {
 
     /// Get the output partitioning of this plan
     fn output_partitioning(&self) -> Partitioning {
-        Partitioning::UnknownPartitioning(1)
+        // because we can have repartitioning using the partition keys
+        // this would be either 1 or more than 1 depending on the presense of
+        // repartitioning
+        self.input.output_partitioning()
     }
 
     fn required_child_distribution(&self) -> Distribution {
-        Distribution::SinglePartition
+        Distribution::UnspecifiedDistribution
```

Comment:
Would this be correct with a window without any `partition by` clause?

In that case I think the required partitions should be 1, as the aggregate function can not be computed only over a part of the data.

Reply:
thanks! this is fixed.